pkg/sdk/alert.go

```diff
@@ -0,0 +1,421 @@
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+type ShowAlertOptions struct {
+	show   bool  `ddl:"static" sql:"SHOW"` //lint:ignore U1000 This is used in the ddl tag
+	Terse  *bool `ddl:"keyword" sql:"TERSE"`
+	alerts bool  `ddl:"static" sql:"ALERTS"` //lint:ignore U1000 This is used in the ddl tag
+
+	// optional
+	Like       *Like   `ddl:"keyword" sql:"LIKE"`
+	In         *In     `ddl:"keyword" sql:"IN"`
+	StartsWith *string `ddl:"parameter,no_equals,single_quotes" sql:"STARTS WITH"`
+	Limit      *int    `ddl:"parameter,no_equals" sql:"LIMIT"`
+}
+
+func (v *Alert) ID() SchemaObjectIdentifier {
+	return NewSchemaObjectIdentifier(v.DatabaseName, v.SchemaName, v.Name)
+}
+
+func (v *Alert) ObjectType() ObjectType {
+	return ObjectTypeAlert
+}
+
+type Alert struct {
+	CreatedOn    time.Time
+	Name         string
+	DatabaseName string
+	SchemaName   string
+	Owner        string
+	Comment      *string
+	Warehouse    string
+	Schedule     AlertSchedule
+	State        string
+	Condition    string
+	Action       string
+}
+
+type AlertSchedule interface {
+	alertSchedule()
+	String() string
+}
+
+func AlertScheduleFromString(s string) (AlertSchedule, bool) {
+	parts := strings.Fields(s)
+	if len(parts) == 2 && parts[len(parts)-1] == "MINUTE" {
+		return AlertScheduleInterval(toInt(parts[0])), true
+	} else if withoutCron := dropUsingCron(parts); len(withoutCron) == 6 {
+		return AlertScheduleCronExpression{
+			Expression: strings.Join(withoutCron[0:len(withoutCron)-1], " "),
+			TimeZone:   withoutCron[len(withoutCron)-1],
+		}, true
+	} else {
+		return nil, false
+	}
+}
+func dropUsingCron(parts []string) []string {
+	if parts[0] == "USING" && parts[1] == "CRON" {
+		return parts[2:]
+	} else {
+		return parts
+	}
+}
+
+type AlertScheduleInterval int
+
+func (interval AlertScheduleInterval) String() string {
+	return fmt.Sprintf("%d MINUTE", interval)
+}
+
+func (AlertScheduleInterval) alertSchedule() {}
+
+type AlertScheduleCronExpression struct {
+	Expression string
+	TimeZone   string
+}
+
+func (expression AlertScheduleCronExpression) String() string {
+	return fmt.Sprintf("USING CRON %v %v", expression.Expression, expression.TimeZone)
+}
+
+func (AlertScheduleCronExpression) alertSchedule() {}
+
+func (v *Alert) IsEnabled() bool {
+	return strings.ToLower(v.State) == "started"
+}
+
+func (v *Alert) IsSuspended() bool {
+	return strings.ToLower(v.State) == "suspended"
+}
+
+type alertDBRow struct {
+	CreatedOn    time.Time `db:"created_on"`
+	Name         string    `db:"name"`
+	DatabaseName string    `db:"database_name"`
+	SchemaName   string    `db:"schema_name"`
+	Owner        string    `db:"owner"`
+	Comment      *string   `db:"comment"`
+	Warehouse    string    `db:"warehouse"`
+	Schedule     string    `db:"schedule"`
+	State        string    `db:"state"` // suspended, started
+	Condition    string    `db:"condition"`
+	Action       string    `db:"action"`
+}
+
+func (row alertDBRow) toAlert() *Alert {
```

Comment:
okay so the point of having separate alertDBRow and an Alert struct is so that we can transform the data to be more user friendly. For example, instead of just keeping state as a string, we could convert. it to an enum

Reply:
ok

Reply:
ok